### PR TITLE
Fix https://github.com/NuGet/Home/issues/225

### PR DESCRIPTION
**Pull request by yishaigalatzer**
Fix https://github.com/NuGet/Home/issues/1547
1. Change the order of uninstall to run the uninstall.ps1 script first (to match NuGet 2)
2. Check that the folder exists before trying to delete it
3. Remove duplicate call to Remove file from the project

### CODE REVIEW

**Comment by yishaigalatzer** [comment]
Need a functional test to install and uninstall jQuery 2.1.4 and make sure scripts folder dissapears.

@danliu

**Comment by yishaigalatzer** [comment]
@emgarten @deepakaravindr @MeniZalzman code review please

File: src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs [hunk 1]
```diff
@@ -403,36 +380,57 @@ private static bool IsSkipAssemblyReferences(INuGetProjectContext nuGetProjectCo
...
                     MSBuildNuGetProjectSystemUtility.GetMostCompatibleGroup(packageTargetFramework, buildFileGroups);
 
-                // TODO: Need to handle References element??
 
-                // Step-4: Raise PackageUninstalling event
-                if (PackageUninstalling != null)
-                {
-                    PackageUninstalling(this, packageEventArgs);
-                }
                 PackageEventsProvider.Instance.NotifyUninstalling(packageEventArgs);
```
Comment (deepakaravindr) [inline src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs:431]: Should this move up and be along with 'PackageUninstalling' event too?

File: src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs [hunk 2]
```diff
@@ -403,36 +380,57 @@ private static bool IsSkipAssemblyReferences(INuGetProjectContext nuGetProjectCo
...
 
+                // Step-2: Execute powershell script - uninstall.ps1
+                var toolItemGroups = packageReader.GetToolItems();
+                var compatibleToolItemsGroup = MSBuildNuGetProjectSystemUtility
+                    .GetMostCompatibleGroup(packageTargetFramework, toolItemGroups);
+
+                if (MSBuildNuGetProjectSystemUtility.IsValid(compatibleToolItemsGroup))
+                {
+                    var uninstallPS1RelativePath = compatibleToolItemsGroup.Items.FirstOrDefault(
+                        p => p.EndsWith(Path.DirectorySeparatorChar + PowerShellScripts.Uninstall, StringComparison.OrdinalIgnoreCase));
```
Comment (deepakaravindr) [inline src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs:405]: Please break long lines (since the lines are moved anyways)

**Comment by deepakaravindr** [comment]
:shipit: (Signed-off with comments)

File: src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs (hunk 2 above)
Comment (yishaigalatzer) [inline src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs:405]: k

**Pull request closed without merging**